oidc-service-impl/src/main/java/io/mosip/esignet/services/OpenIdConnectServiceImpl.java

```diff
@@ -51,6 +51,10 @@ public String getUserInfo(String accessToken) throws EsignetException {
             if(transaction == null)
                 throw new NotAuthenticatedException();
 
+            if(transaction.isDpopBoundAccessToken() && !tokenService.isValidDpopServerNonce(dpopHeader, transaction)) {
+                tokenService.generateAndStoreNewNonce(accessTokenHash, Constants.USERINFO_CACHE);
```

Comment:
After generating new nonce, we should fail the request. So that RP can send new request with the nonce